go/vt/tabletserver/query_executor.go

```diff
@@ -295,25 +295,33 @@ func (qre *QueryExecutor) execDDL() (*sqltypes.Result, error) {
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "DDL is not understood")
 	}
 
-	conn, err := qre.tsv.te.txPool.LocalBegin(qre.ctx)
+	if qre.transactionID != 0 {
```

Comment:
Can you explain a little more about why you think we should do that? I'm not sure if my original use-case was lost in the commit stream, but here's what I see from my side:

Liquibase (a major java schema migration manager) converts a user-specified "changeset" into a bundle of queries. In many cases that results in just 1 query, like an ALTER/CREATE/etc. But for some changesets, that can be multiple queries.  Here's an example:

```
<changeSet id="Populate initial values and make not null" author="bbeaudreault" context="job">
        <addNotNullConstraint tableName="todo" columnName="list_id" defaultNullValue="1" columnDataType="int"/>
    </changeSet>
```

This is a reasonably common example: Adding a not-null constraint on a column, and first setting the default value on all existing rows. The way to handle this is to do 2 queries, wrapped in a transaction:

```
2017-03-02T20:34:26.971972Z        40 Query     begin
2017-03-02T20:34:26.972148Z        40 Query     update vttest.todo set list_id = '1' where list_id is null/* vtgate:: filtered_replication_unfriendly */
2017-03-02T20:34:27.004152Z        40 Query     ALTER TABLE vttest.todo MODIFY list_id INT NOT NULL
2017-03-02T20:34:27.061314Z        40 Query     commit
```

The above works great with the changes in this PR. However, previously the queries would come in like this:

```
begin
update vttest.todo set list_id = '1' where list_id is null/* vtgate:: filtered_replication_unfriendly */
begin
ALTER TABLE vttest.todo MODIFY list_id INT NOT NULL
--- unreachable:
commit
commit
```

You may be right that all DDLs have an implicit commit -- is that a MySQL thing, or a vitess thing? If Vitess, is there a reason that's the case? Some other way we can do it?

The way I see it is this is a reasonable use-case, and one used by a major tool that is very popular in the java community. So there should be a way to support it, not throw an exception.

One option, which would solve this case but not sure about all other changesets:  We can still do the implicit commit, but only call LocalBegin if we're not already in a transaction.

Thoughts?